build_runner/bin/create_merged_dir.dart

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:args/args.dart';
+import 'package:build/build.dart';
+import 'package:path/path.dart' as p;
+
+import 'package:build_runner/src/asset/build_cache.dart';
+import 'package:build_runner/src/asset/file_based.dart';
+import 'package:build_runner/src/asset_graph/graph.dart';
+import 'package:build_runner/src/asset_graph/node.dart';
+import 'package:build_runner/src/package_graph/package_graph.dart';
+import 'package:build_runner/src/util/constants.dart';
+
+AssetGraph assetGraph;
+PackageGraph packageGraph;
+
+Future main(List<String> args) async {
+  stdout.writeln(
+      'Warning: this tool is unsupported and usage may change at any time, '
+      'use at your own risk.\n\n'
+      'This tool also assumes you are using the `writeToCache=true` option.');
+
+  var parsedArgs = argParser.parse(args);
+
+  var scriptPath = parsedArgs['script'] as String;
+  if (scriptPath == null) {
+    throw new ArgumentError.notNull('--script');
+  }
+
+  var outputDir = new Directory(parsedArgs['output-dir'] as String);
+  if (outputDir.existsSync()) {
+    stdout.writeln('Deleting existing output dir `$outputDir`');
+    outputDir.deleteSync(recursive: true);
+  }
+  outputDir.createSync(recursive: true);
+
+  var scriptFile = new File(scriptPath);
+  if (!scriptFile.existsSync()) {
+    throw new ArgumentError(
```

Comment:
Can we move this and other checks above where we delete the old output?

Reply:
done